File: 2302-Count Subarrays With Score Less Than K.py
class Solution(object):
    def countSubarrays(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        res = 0
        left = 0
        
        running = 0
        for i in range(len(nums)):
            running += nums[i]
            while left <= i and running*(i-left+1) >= k:
                running = running - nums[left]
                left += 1
            res += (i-left+1)
            
        return res

File: C++/2302-Count Subarrays With Score Less Than K.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, long long k) {
        int N = nums.size(), left = 0;
        long long res = 0, sum = 0;

        for (int i=0; i<N; i++) {
            sum += nums[i];
            while (sum*(i-left+1) >= k) {
                sum -= nums[left];
                left++;
            }
            res += i-left+1;
        }

        return res;
    }
};